### ISSUE TITLE
Memory leak on repeated auth calls

### ISSUE THREAD

**Issue by abb-sandeeprao** [AzureAD/microsoft-authentication-library-for-js#7401]
### Core Library

MSAL.js (@azure/msal-browser)

### Core Library Version

14.11.0

### Wrapper Library

Not Applicable

### Wrapper Library Version

None

### Public or Confidential Client?

Public

### Description

For the given configuration, we are seeing that the memory is gradually increasing with Strings, and concatenated Strings from the PerformanceClient instance.

![image](https://github.com/user-attachments/assets/5a6b04aa-cfdc-4139-b859-a2c5e2642c75)

Identity Provider: Identityserver4

### Network Trace (Preferrably Fiddler)

- [ ] Sent
- [ ] Pending

### MSAL Configuration

```javascript
{
    "auth": {
        "clientId": "ABBThinClient",
        "authority": "https://udayauthserver1:442/auth-server",
        "knownAuthorities": [
            "https://udayauthserver1:442/auth-server"
        ],
        "cloudDiscoveryMetadata": "",
        "authorityMetadata": "",
        "redirectUri": "",
        "postLogoutRedirectUri": "",
        "navigateToLoginRequestUrl": true,
        "clientCapabilities": [],
        "protocolMode": "OIDC",
        "OIDCOptions": {
            "serverResponseType": "fragment",
            "defaultScopes": [
                "openid",
                "profile",
                "offline_access"
            ]
        },
        "azureCloudOptions": {
            "azureCloudInstance": "none",
            "tenant": ""
        },
        "skipAuthorityMetadataCache": false,
        "supportsNestedAppAuth": false
    },
    "cache": {
        "cacheLocation": "localStorage",
        "temporaryCacheLocation": "sessionStorage",
        "storeAuthStateInCookie": false,
        "secureCookies": false,
        "cacheMigrationEnabled": true,
        "claimsBasedCachingEnabled": true
    },
    "system": {
        "tokenRenewalOffsetSeconds": 60,
        "preventCorsPreflight": false,
        "networkClient": {},
        "navigationClient": {},
        "loadFrameTimeout": 0,
        "windowHashTimeout": 60000,
        "iframeHashTimeout": 10000,
        "navigateFrameWait": 0,
        "redirectNavigationTimeout": 30000,
        "asyncPopups": false,
        "allowRedirectInIframe": false,
        "allowNativeBroker": false,
        "nativeBrokerHandshakeTimeout": 2000,
        "pollIntervalMilliseconds": 30,
        "loggerOptions": {
            "logLevel": 2,
            "piiLoggingEnabled": false
        }
    },
    "telemetry": {
        "application": {
            "appName": "",
            "appVersion": ""
        },
        "client": {}
    }
}
```

### Relevant Code Snippets

```javascript
import {
  PublicClientApplication
} from "@azure/msal-browser";

this.client = new PublicClientApplication(<config>);

const { accessToken } = await this.client.acquireTokenSilent({
        account: ...,
        scopes: ...,
      });
```

### Reproduction Steps

1. Connect web application to Auth Server.
2. Make repeated calls to getAuthToken() API
3. Take repeated heap dumps every 3 or 4 hours.
4. Seeing that the memory keeps increasing.

### Expected Behavior

Memory should not move up and remain stationary.

### Identity Provider

Other

### Browsers Affected (Select all that apply)

Chrome, Edge

**Comment by konstantin-msft** [comment]
@abb-sandeeprao Thanks for reporting this. I have added this issue to my short-term to-do list.

**Comment by konstantin-msft** [comment]
@abb-sandeeprao can you please check that you are running latest version of msal.js? `staticFieldsByCorrelationId` map was removed from `performanceClient` in Feb 2023. Please see [the corresponding PR](https://github.com/AzureAD/microsoft-authentication-library-for-js/commit/b03430c5841e5caad33d0b928b160af9f1535708#diff-5c2e3cc078d85a3fa0eb4b4596ecee70cc03689d6ac7ab96c69b8628fd8e055fL40). 
I wasn't able to find memory leaks in perf client in modern version of msal.js.

**Comment by abb-sandeeprao** [comment]
Thanks for looking into this @konstantin-msft

We are currently using @azure/msal-browser": "^3.16.0. Should I try upgrading to 3.26.1?

**Comment by konstantin-msft** [comment]
It's already fixed in 3.16.0+. Can you please make sure you are not using multiple versions of msal.js in your app?

**Comment by abb-sandeeprao** [comment]
We have a single version of MSAL. We are currently switching in between apps hence, had posted a different version initially.

Here is the scenario I am facing.

We are continuously loading live data in the screen, which requires continuous calls to our backend, each calling get Auth(). During this scenario, the PerformanceClient is allocating and deallocating memory, but it is gradually increasing in nature. This is resulting in an application crash after the memory rises to more than 3 GB in 3 days time.

Is there a way to turn off Performance Client? If not, can you provide the option?
